src/emulator/storage/cloudFunctions.ts

```diff
@@ -69,6 +85,22 @@ export class StorageCloudFunctions {
       }, // bucket
       data: objectMetadataPayload,
     });
+    /** Cloud Event */
+    const ceAction = STORAGE_V2_ACTION_MAP[action];
+    const data = { ...objectMetadataPayload };
+    delete (data as any).acl; // remove this field from cloud events
+    const ce = {
+      specVersion: 1,
+      type: `google.cloud.storage.object.v1.${ceAction}`,
+      source: `//storage.googleapis.com/projects/_/buckets/${objectMetadataPayload.bucket}/objects/${objectMetadataPayload.name}`,
+      data,
+    };
+    const cloudEvent: string = JSON.stringify({
+      eventType: `google.cloud.storage.object.v1.${ceAction}`, // need eventType for multicast triggerId
```

Comment:
Hmm this implementation feels a bit brittle since `eventType` isn't a thing in a cloudevent payload. It also makes  an implicit assumption that the name of the event will always be different between legacy and cloudevent. That is actually, probably true, but it's a subtle assumption that's worth being explicit (either in code or in implementation).

Another way to implement this change would be to teach the multicast handler to understand the difference between cloudevent and legacy event payloads (either by having different endpoint, having a wrapper request body e.g. `{ eventFormat: "cloudevent", eventBody: body }`, etc. Do you think that is going to work?

Reply:
Discussed offline - we will probably use the value of the `Content-Type` header to differentiate cloudevent vs legacy payload.

Also, I think this already works w/o any modification, but I'd make sure that trigger registration where we construct the key for the `multicastTriggers` object still works:

https://github.com/firebase/firebase-tools/blob/e4b9b938716b6679e132a13a82fd7f16bdda5804/src/emulator/functionsEmulator.ts#L695-L703